webclient/http1/src/main/java/io/helidon/webclient/http1/Http1CallChainBase.java

```diff
@@ -510,7 +512,12 @@ public int read() {
             if (finished || currentBuffer == null) {
                 return -1;
             }
-            return currentBuffer.read();
+            int read = currentBuffer.read();
+            //If we read fully the current chunk, and we are not asked to read again
+            //(such as when considered fully processed),
+            //we will have the last chunk not processed and hanging
+            ensureBuffer();
```

Comment:
This is wrong, as it pre-fetches data that may not be needed.
An input stream cannot be left "hanging" as the comment says - it must be fully read and closed. 
If there is a bug that we do not read the last zero length chunk on close, it must be fixed, but this does not seem to be the right place.